src/nupic/algorithms/Connections.cpp

```diff
@@ -307,49 +364,37 @@ bool Connections::compareSegments(Segment a, Segment b) const {
 
 vector<Synapse>
 Connections::synapsesForPresynapticCell(CellIdx presynapticCell) const {
-  if (synapsesForPresynapticCell_.find(presynapticCell) ==
-      synapsesForPresynapticCell_.end())
-    return vector<Synapse>{};
-
-  return synapsesForPresynapticCell_.at(presynapticCell);
-}
-
-Synapse Connections::minPermanenceSynapse_(Segment segment) const {
-  // Use special EPSILON logic to compensate for floating point differences
-  // between C++ and other environments.
-
-  bool found = false;
-  Permanence minPermanence = std::numeric_limits<Permanence>::max();
-  Synapse minSynapse;
-
-  for (Synapse synapse : segments_[segment].synapses) {
-    if (synapses_[synapse].permanence < minPermanence - EPSILON) {
-      minSynapse = synapse;
-      minPermanence = synapses_[synapse].permanence;
-      found = true;
-    }
-  }
-
-  NTA_CHECK(found);
-
-  return minSynapse;
+  vector<Synapse> all(
```

Comment:
might call potentialAndConnectedSyns (?)

Reply:
I can't change the name `Connections::synapsesForPresynapticCell` since its part of the public facing API.

Reply:
no, I was talking about the `vector<Synapse> all` since it (kind of) conflicts with all as function in c++.